### ISSUE TITLE
noImplicitThis + noUnusedLocals = impossible to use together

### ISSUE THREAD

**Issue by pocesar** [Microsoft/TypeScript#9567]
**TypeScript Version:**  nightly

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
TestSchema
.static('pages', function(this: IModelTest, query: mongoose.Query<IModelTest>){
  return this.paginate(query);
});
```

**Expected behavior:**

noUnusedLocals should ignore `this` typing, when noImplicitThis is enabled

**Actual behavior:** 

`this: IModelTest` is showing as an unused variable, making it impossible to use both options at the same time

**Comment by DickvdBrink** [comment]
Fixed by #9557?

**Comment by pocesar** [comment]
it does, thanks!
